### ISSUE TITLE
add zkSync Era

### ISSUE THREAD

**Issue by githubdoramon** [WalletConnect/walletconnect-monorepo#2078]
**Adding a new chain to the Explorer**
To get a new chain added to the Explorer, you will need to submit the following:

1. JSON-RPC spec for Wallets - Not needed. Already available at rpc-proxy repo
   **GitHub Link**: https://github.com/WalletConnect/rpc-proxy/
2. Write a [CASA namespace spec](https://github.com/ChainAgnostic/namespaces) if not already available for this namespace.
   **GitHub Link**: Not needed
3. **namespaces**: eip:155
4. **chains**:
  zkSync Era testnet: eip155:280
  zkSync Era mainnet: eip155:324
5. **RPC endpoints**
   280: https://zksync2-testnet.zksync.dev
   324: https://zksync2-mainnet.zksync.io
6. **[SLIP-0044](https://github.com/satoshilabs/slips/blob/master/slip-0044.md) coin type**:
   804: 0x80000324 -> Caveat here is, this coin type was created for coinbase, but zkSync Era runs on ETH

**Comment by bkrem** [comment]
Thanks for adding zkSync Era 🙌 

LGTM, but will wait for greenlight from @bumblefudge to avoid unaddressed concerns. 

Since this is also `eip155` I'm hoping this is straightforward though.

**Comment by bumblefudge** [comment]
There was [an earlier conversation](https://github.com/ChainAgnostic/CAIPs/issues/100) about how to address zkSync v1 addresses, but am I right in _assuming_ this is just for zkSync v2 (as the RPC endpoints would imply) and there's no legacy compat requirement?

I also _assume_ there is NOT, but if there is anything fundamentally different about CAIP:10 or CAIP:19 on zkSync era (i.e. values of X that would be valid for `eip155:1:X` but not for `eip155:324:X` for either CAIP-10 or CAIP-19), it might be worth adding a section to [/namespaces/eip155/caip10.md](https://github.com/chainAgnostic/namespaces/tree/main/eip155/caip10.md) and/or [/namespaces/eip155/caip19.md](https://github.com/chainAgnostic/namespaces/tree/main/eip155/caip19.md) - glad to help if so.  For example, if only [supported tokens](https://zkscan.io/explorer/tokens/) would be valid addressed as `eip155:324:X`, and all other ERC20s should be addressed as `eip155:1:X` even in a zkSync Era context, then a one-sentence mention of this chain-ID-specific constraint would might be a good heads-up for wallet (or more likely, dapp) implementers with no ZKSync experience trying to implement multichain through WC rather than manually. Particularly if that one sentence linked to this authoritative registry, ideally the machine-readable one that generates the human-readable link above, so that, e.g., a dashboard dapp could validate before presenting invalid links or information to an end-user. This isn't a hard requirement of WC registration, but it might still be worthwhile as a devrel/footgun-protection measure :D

As for the SLIP-0044 registration, I'm confused-- does Era still use the SLIP-registered derivation path (i.e. do wallets actually derive keys this way for anything) or does Era just use Eth natively now?  I guess this relates back to the legacy question

**Comment by githubdoramon** [comment]
@bumblefudge sry about the delay interacting here

Some answers (but would definitely need your help due to lack of deeper knowledge on CAIPs and SLIPs.

- no legacy compat requirement between zkSync 1.0 (now zkSync Lite) and 2.0 (now zkSync Era) - You are correct... they are 2 totally different chains
- Given Era is EVM compatible, any EVM address (eip155:1:X) will be valid on Era (eip155:324:x). That being said, for contracts, address derivation is different, so same contract will have different address on L1 and L2 (but I guess this doesn't impact at all the current conversation right?)
- Supported tokens - the link you provided is for zkSync Lite... For Era ERC20 will be EVm compatible as well, but as I mentioned, they will have different addresses from L1. Not sure if this is actually related to your question, or if I might have not understood it. Please let me know.
- On the SLIP-0044 - I understand very little about this, but I guess the coin added there is indeed related to zkSync Lite. zkSync Era uses ETH as gas token (of course, our version of L2 ETH, which is "bridgeable" for the L1 one. Please also give me some light on how to answer this question properly :)

Thanks a lot for the help so far.

**Comment by githubdoramon** [comment]
Correcting one of my questions/answers @bumblefudge  ... the coin type on slip-044 is actually for zkSync Era -> It was initially requested by Argent. But from my new understanding, this is not related to ETH being the native token no?

**Comment by bumblefudge** [comment]
Hehe, now I'm even more confused about the SLIP-44 registration! Is it an argent-specific ERC20 or is it a private form that the Eth gets wrapped to for use on Era?  Any more info (or tagging in folks from Argent who might know) appreciated!  I don't think the Slip-44 thing should be a blocker, but I am always eager to do more documentation than the minimum, and would be interested to know more in case there's an annotation or a code-comment worth inserting somewhere to help more wallets and dapps support you.

Address derivation being different is fine, I think that's the same on many L2s, so no one expects the address to be the same even if the ABI hash is.  It would only be significant if deployed contracts were *addressed* differently, i.e., do the addresses *look* different, i.e. do you need a different regex to validate them?  I assume not, in which case there's nothing worth documenting there :D

My mistake on the "support tokens" list, I didn't notice I was on the old docs! You understood just fine, it was me who was lost.

In summary: you're good to go!  Thanks for answering my questions

**Comment by githubdoramon** [comment]
Many thx @bumblefudge 

- Regarding SLIP-0044 - From my understanding, it has nothing to do with ETH or any ERC20... it just has a bad name (cointype - where it is actually related to the chain itself).
- Address derivation - nothing different indeed, same "type" of address :)

**Comment by Cali93** [comment]
The references to zkSync Era have been added on the Explorer so you're all set! 🚀
